passio-ui-module/src/main/java/ai/passio/nutrition/uimodule/ui/foodcreator/FoodCreatorViewModel.kt

```diff
@@ -320,17 +324,17 @@ class FoodCreatorViewModel : BaseViewModel() {
 //        nutritionFacts.servingSize
 //        nutritionFacts.sugarAlcohol
 
-        requiredNutritionFacts.setValue(REF_CARBS_ID, nutritionFacts.carbs ?: 0.0)
-        requiredNutritionFacts.setValue(REF_CALORIES_ID, nutritionFacts.calories ?: 0.0)
-        requiredNutritionFacts.setValue(REF_PROTEIN_ID, nutritionFacts.protein ?: 0.0)
-        requiredNutritionFacts.setValue(REF_FAT_ID, nutritionFacts.fat ?: 0.0)
+        requiredNutritionFacts.setValue(REF_CARBS_ID, nutritionFacts.carbs?.div(ratio) ?: 0.0)
+        requiredNutritionFacts.setValue(REF_CALORIES_ID, nutritionFacts.calories?.div(ratio) ?: 0.0)
+        requiredNutritionFacts.setValue(REF_PROTEIN_ID, nutritionFacts.protein?.div(ratio) ?: 0.0)
+        requiredNutritionFacts.setValue(REF_FAT_ID, nutritionFacts.fat?.div(ratio) ?: 0.0)
```

Comment:
With the new PassioNutrients constructor that accepts the weight, the constructor itself calculates the ratio so you don't have to do it on the UI side. Also, a number of other nutrients are left commented out?

Reply:
@marinPassio
I'm applying reverse logic when editing data. 

For example, if the weight is 50 grams and the carbs value is 10 grams, we pass the carbs to the `PassioNutrients` constructor with the weight in grams. The SDK then adjusts the carbs based on a 100-gram weight. 

Here's how it works: This calculation is handled on the SDK side.
```kotlin
val ratio = 100 / weight.gramsValue() // 100 / 50 = 2
val carbsValue = carbs?.times(ratio) // 10 * 2 = 20
```

Please suggest me if I'm doing anything wrong.
On the app side, I apply the reverse formula to display values according to the actual weight:
```kotlin
val ratio = 100 / weight.gramsValue() // 100 / 50 = 2
val carbsValue = carbs?.div(ratio) // 20 / 2 = 10
```

Reply:
> a number of other nutrients are left commented out?

I'm not currently receiving values for the commented nutrients from the nutrition facts scanning. I've kept these commented out as placeholders in case we decide to include them in the nutrition facts in the future.